fee/src/lib.rs

```diff
@@ -0,0 +1,40 @@
+use {solana_sdk::fee::FeeDetails, solana_svm_transaction::svm_message::SVMMessage};
+
+/// Calculate fee for `SanitizedMessage`
+pub fn calculate_fee(
+    message: &impl SVMMessage,
+    zero_fees_for_test: bool,
+    lamports_per_signature: u64,
+    prioritization_fee: u64,
+    remove_rounding_in_fee_calculation: bool,
+) -> u64 {
+    calculate_fee_details(
+        message,
+        zero_fees_for_test,
+        lamports_per_signature,
+        prioritization_fee,
+        remove_rounding_in_fee_calculation,
+    )
+    .total_fee()
+}
+
+pub fn calculate_fee_details(
+    message: &impl SVMMessage,
```

Comment:
I wonder if giving `num_total_signatures: u32` instead of `message: &impl SVMMessage` would make the API concise, and less depends? 

Also thinking about what we chatted early that to reduce iteration of instructions, `num_total_signatures` can be retrieved once and cached.

Reply:
Could simplify but given there was the recent confusion on `num_signatures()` (now renamed to `num_total_signatures()` and `signatures.len()` I feel it could lead to us screwing up a call-site somewhere.

Thinking we leave it as for now, and the `SVMMessage` will abstract away our caching of the value.